tools/flakes/detect_flakes.py

```diff
@@ -13,92 +13,251 @@
+GH_ISSUES_URL = 'https://api.github.com/repos/grpc/grpc/issues'
+KOKORO_BASE_URL = 'https://kokoro2.corp.google.com/job/'
+
+def gh(url, data=None):
+  request = urllib2.Request(url, data=data)
+  assert TOKEN
+  request.add_header('Authorization', 'token {}'.format(TOKEN))
+  if data:
+    request.add_header('Content-type', 'application/json')
+  response = urllib2.urlopen(request)
+  if 200 <= response.getcode() < 300:
+    return json.loads(response.read())
+  else:
+    raise ValueError('Error ({}) accessing {}'.format(
+        response.getcode(), response.geturl()))
+
+
+def create_gh_issue(title, body, labels):
+  data = json.dumps({'title': title,
+                     'body': body,
+                     'labels': labels})
+  response = gh(GH_ISSUES_URL, data)
+  issue_url = response['html_url']
+  print('Issue {} created for {}'.format(issue_url, title))
 
 
-def get_flaky_tests(days_lower_bound, days_upper_bound, limit=None):
-  """ period is one of "WEEK", "DAY", etc.
-  (see https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#date_add). """
+def build_kokoro_url(job_name, build_id):
+  job_path = '{}/{}'.format('/job/'.join(job_name.split('/')), build_id)
+  return KOKORO_BASE_URL + job_path
 
+
+def create_issues(new_flakes):
+  for test_name, results_row in new_flakes.items():
+    poll_strategy, job_name, build_id, timestamp = results_row
+    url = build_kokoro_url(job_name, build_id)
+    title = 'New Flake: ' + test_name
+    body = '- Test: {}\n- Poll Strategy: {}\n- URL: {}'.format(
+        test_name, poll_strategy, url)
+    labels = ['infra/New Flakes']
+    create_gh_issue(title, body, labels)
+
+
+def print_table(table, format):
+  for test_name, results_row in table.items():
+    poll_strategy, job_name, build_id, timestamp = results_row
+    ts = int(float(timestamp))
+    # TODO(dgq): timezone handling is wrong. We need to determine the timezone
+    # of the computer running this script.
+    human_ts = datetime.datetime.utcfromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S UTC')
+    full_kokoro_url = build_kokoro_url(job_name, build_id)
+    if format == 'human':
+      print("\t- Test: {}, Polling: {}, Timestamp: {}, url: {}".format(
+          test_name, poll_strategy, human_ts, full_kokoro_url))
+    else:
+      assert(format == 'csv')
+      print("{},{},{}".format(test_name, ts, human_ts, full_kokoro_url))
+
+Row = namedtuple('Row', ['poll_strategy', 'job_name', 'build_id', 'timestamp'])
+def get_flaky_tests(from_date, to_date, limit=None):
+  """Return flaky tests for date range (from_date, to_date], where both are
+  strings of the form "YYYY-MM-DD" """
   bq = big_query_utils.create_big_query()
   query = """
+#standardSQL
 SELECT
-  REGEXP_REPLACE(test_name, r'/\d+', '') AS filtered_test_name,
+  RTRIM(LTRIM(REGEXP_REPLACE(filtered_test_name, r'(/\d+)|(bins/.+/)|(cmake/.+/.+/)', ''))) AS test_binary,
```

Comment:
+1 to improving the flake detection feature, I think that's very useful.

Reply:
I've further improved and simplified the logic. Now it's a single (yet huge) SQL query, the same one used by the dashboard, so both should display consistent results.